tapir/coop/services/ResignMemberService.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+
+from tapir.accounts.models import TapirUser
+from tapir.coop.models import ResignedMembership, ShareOwnership
+from tapir.coop.forms import MembershipCancelForm
+from tapir.shifts.models import (
+    ShiftAttendanceTemplate,
+    ShiftAttendance,
+    ShiftAccountEntry,
+)
+from tapir.shifts.services.frozen_status_service import FrozenStatusService
+from tapir.shifts.models import ShiftUserData, ShiftAttendanceMode
+
+from tapir.utils.shortcuts import get_timezone_aware_datetime
+
+
+class ResignMemberService:
+    @staticmethod
+    def update_shifts_and_shares(
+        form: MembershipCancelForm, member: ResignedMembership
```

Comment:
I would rename `member` to `resignation` or something similar, since it's not really the member that is contained in this variable.